spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -236,21 +239,55 @@ public void doWith(Method method) throws IllegalArgumentException, IllegalAccess
 				}
 			}
 		});
-		if (classLevelListener != null) {
-			processMultiMethodListener(classLevelListener, multiMethods, bean, beanName);
+		if (hasClassLevelListeners) {
+			processMultiMethodListeners(classLevelListeners, multiMethods, bean, beanName);
 		}
 		return bean;
 	}
 
-	private void processMultiMethodListener(RabbitListener classLevelListener, List<Method> multiMethods, Object bean,
-			String beanName) {
+	/*
+	 * AnnotationUtils.getRepeatableAnnotations does not look at interfaces
```

Comment:
It doesn't matter how `JmsListener` behaves there, I think that some generic approach would be good to have in the core of cores :smile: 

CC @sbrannen